service/src/main/java/com/epam/esm/validator/UserValidator.java

```diff
@@ -0,0 +1,77 @@
+package com.epam.esm.validator;
+
+import com.epam.esm.constants.ApplicationConstants;
+import com.epam.esm.dao.UserDao;
+import com.epam.esm.entity.User;
+import com.epam.esm.exceptions.UserException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+/**
+ * Implementation of {@link EntityValidator} for User entity.
+ *
+ * @since 2.0
+ */
+@Component
+public class UserValidator implements EntityValidator<User> {
+
+    private final UserDao userDao;
+
+    @Autowired
+    public UserValidator(UserDao userDao) {
+        this.userDao = userDao;
+    }
+
+
+    /**
+     * This method attempts to get an User entity from db by it's id.
+     *
+     * @param userId id of the User entity.
+     * @return User entity.
+     * @throws UserException when there is no entity with given id in db.
+     * @since 2.0
+     */
+    public User validateAndFindByIdIfExist(long userId) {
+        return userDao.findById(userId)
```

Comment:
Validators should only validate some params or entities or results and etc.
There should be no getting entities from the database for validation. This must be done in the appropriate service.

Reply:
Commit 9645c6a.